### PR TITLE
refactor: deepen architecture (StatusView seam, pure PR parse, module split)

### PR DESCRIPTION
**Pull request by karbassi**
Three behavior-preserving deepening refactors, each moving a test surface from "spawn the binary" to "call an interface." Output is byte-identical to `main`.

## A — Split gathering from rendering (`c714bdb`)
`RenderContext::new` did git stats, PR fetch, and a hostname syscall in its constructor, trapping the pure 22-arm `render_component` tree behind a process spawn. Lifted all I/O into `gather()` returning a plain `StatusView`; `render_component`/`write_rows` are now pure over it.

## B + C — Pure PR parse + cache codec (`0c19558`)
The gh shell-script path and native HTTP path each parsed responses and hand-wrote the cache format; check-status logic existed twice. Extracted pure `compute_check_status`, `parse_pr_payload`, and `encode/decode_pr_cache` (one home for the `timestamp\nbranch\npayload` format, mirroring `MmapCache`). The detached gh refresh script is the one remaining hand-rolled writer (noted inline).

Skipped a `PrSource` trait: gh-vs-HTTP is a compile-time/runtime pick at one call site, never substituted in a test — a trait there is ceremony, not a seam.

## D — Module split (`780f2a2`)
Pure code move along the seams the prior commits created: `git.rs` (GitRepo + mmap cache), `pr.rs` (codec, parse, fetch adapters), `render.rs` (StatusView, gather, components). `main.rs` 2218 → 706 lines; cross-module items are `pub(crate)`.

## Testing
- 15 new pure unit tests covering previously-uncovered logic (PR check-status edge cases, cache codec round-trips, render arms)
- `cargo test` — 64 unit + 25 integration + 5 lib, all green
- `cargo clippy --all-targets -- -D warnings` clean
- `cargo fmt --check` clean
- Release binary output byte-identical to `main`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/render.rs [hunk 1]
```diff
@@ -0,0 +1,526 @@
...
+        .and_then(|p| Path::new(p).file_name())
+        .map(|n| n.to_string_lossy().into_owned())
+        .unwrap_or_default();
+
+    let home = get_home();
+    let display_cwd = if !home.is_empty() && current_dir.starts_with(home) {
+        format!("~{}", &current_dir[home.len()..])
+    } else {
+        current_dir.to_string()
+    };
```
[the remark this answers is not in the document]
Comment (karbassi) [reply src/render.rs]: Fixed in 63af01b: extracted a pure `collapse_home` that only shortens on an exact HOME match or a real `/`-boundary sub-path, so `/home/alex` no longer becomes `~ex` under `HOME=/home/al`. Added 5 unit tests covering the boundary cases.

File: src/render.rs [hunk 2]
```diff
@@ -0,0 +1,566 @@
...
+        return current_dir.to_string();
+    }
+    if current_dir == home {
+        return "~".to_string();
+    }
+    if let Some(rest) = current_dir.strip_prefix(home)
+        && rest.starts_with('/')
+    {
+        return format!("~{rest}");
+    }
```
[the remark this answers is not in the document]
Comment (karbassi) [reply src/render.rs:64]: Fixed in a0f9a92: accept both `/` and `\` as the sub-path boundary so Windows `%USERPROFILE%` paths collapse, with a test covering backslash boundary + prefix-only rejection.

File: src/render.rs [hunk 3]
```diff
@@ -0,0 +1,566 @@
...
+        "pr_state" => {
+            let pr = view.pr.as_ref()?;
+            let state_lower = pr.state.to_lowercase();
+            let color = match state_lower.as_str() {
+                "open" => TN_GREEN,
+                "merged" => TN_PURPLE,
+                "closed" => TN_RED,
+                _ => TN_GRAY,
+            };
+            Some(format!("{color}{state_lower}{RESET}"))
```
[the remark this answers is not in the document]
Comment (karbassi) [reply src/render.rs:282]: Fixed in aa946b4: `pr_state` now returns None when the state is empty/whitespace, so no bare ANSI codes leak into the row. Added a test.

File: src/pr.rs [hunk 4]
```diff
@@ -0,0 +1,675 @@
...
+        Some(c) if !c.is_empty() => c,
+        _ => return String::new(),
+    };
+
+    let is_passing = |s: &str| {
+        matches!(
+            s.to_ascii_uppercase().as_str(),
+            "SUCCESS" | "SKIPPED" | "NEUTRAL"
+        )
+    };
```
[the remark this answers is not in the document]
Comment (karbassi) [reply src/pr.rs:117]: Fixed in bf1d537: switched to `eq_ignore_ascii_case` against the literals, dropping the per-conclusion `String` allocation. Covered by the existing case-insensitive test.

File: src/render.rs [hunk 5]
```diff
@@ -0,0 +1,603 @@
...
+    if current_dir == home {
+        return "~".to_string();
+    }
+    if let Some(rest) = current_dir.strip_prefix(home)
+        && (rest.starts_with('/') || rest.starts_with('\\'))
+    {
+        return format!("~{rest}");
+    }
+    current_dir.to_string()
+}
```
[the remark this answers is not in the document]
Comment (karbassi) [reply src/render.rs:66]: Fixed in b8c5cc0: `collapse_home` now trims a trailing `/` or `\` from HOME before comparing, so "/home/me/" and "C:\\Users\\me\\" collapse correctly. Added a test.

File: src/pr.rs [hunk 6]
```diff
@@ -0,0 +1,674 @@
...
+// ----------------------------------------------------------------------------
+// PR cache codec — the single home for the on-disk format.
+//
+// Format: `timestamp\nbranch\npayload`, where payload is a JSON blob, the
+// `NO_PR` negative-cache marker, or an `ERROR:...` marker. Every Rust writer
+// goes through `encode_pr_cache`; the one exception is the detached gh refresh
+// shell script (`spawn_pr_refresh_gh`), which builds the same layout via printf
+// because it runs in a separate process.
+// ponytail: keep this format dead simple (line-delimited); switch to a struct +
+// serde only if a field ever needs escaping.
```
[the remark this answers is not in the document]
Comment (karbassi) [reply src/pr.rs]: Fixed in 3261813: reworded to `NOTE:`.

File: src/pr.rs [hunk 7]
```diff
@@ -0,0 +1,674 @@
...
+
+    // Script logic:
+    // 1. Run gh pr view and capture stdout/stderr separately
+    // 2. If gh succeeds with JSON output -> write PR data
+    // 3. If gh fails with "no pull requests" message -> write NO_PR (legitimate no PR)
+    // 4. If gh fails for other reasons -> write ERROR (don't negative cache)
+    // 5. Atomic rename temp file to cache file
+    // Uses trap with $0 for cleanup to avoid quoting issues with shell_escape
+    // ponytail: this printf must mirror encode_pr_cache's `timestamp\nbranch\npayload`
+    // layout by hand — it runs in a detached process and can't call back into Rust.
```
[the remark this answers is not in the document]
Comment (karbassi) [reply src/pr.rs]: Fixed in 3261813: reworded to `NOTE:`.
